std/src/main/scala/algebra/std/set.scala

```diff
@@ -1,17 +1,31 @@
 package algebra
 package std
 
+import algebra.lattice.Lattice
 import algebra.ring.Semiring
 
 package object set extends SetInstances
 
 trait SetInstances {
-  implicit def setEq[A]: Eq[Set[A]] = new SetEq[A]
+  implicit def setLattice[A]: Lattice[Set[A]] = new SetLattice[A]
+  implicit def setPartialOrder[A]: PartialOrder[Set[A]] = new SetPartialOrder[A]
   implicit def setSemiring[A] = new SetSemiring[A]
 }
 
-class SetEq[A] extends Eq[Set[A]] {
-  def eqv(x: Set[A], y: Set[A]): Boolean = x == y
+class SetLattice[A] extends Lattice[Set[A]] {
+  def join(lhs: Set[A], rhs: Set[A]): Set[A] = lhs.union(rhs)
+
+  def meet(lhs: Set[A], rhs: Set[A]): Set[A] = lhs.intersect(rhs)
+}
+
+class SetPartialOrder[A] extends PartialOrder[Set[A]] {
```

Comment:
Good point, I was wondering if I was making too many passes over the `Set`s - change has been made.